doc/cpp-target.md

```diff
@@ -102,6 +102,11 @@ In order to create a static lib in Visual Studio define the `ANTLR4CPP_STATIC` m
 
 For gcc and clang it is possible to use the `-fvisibility=hidden` setting to hide all symbols except those that are made default-visible (which has been defined for all public classes in the runtime).
 
+### Compile Aspects
+When compiling generated files, you can configure a compile option according to your needs (also optional):
+
+* A **thread local DFA macro**: Add `-DANTLR4_USE_THREAD_LOCAL_CACHE=1` to the compilation options will enable using thread local DFA cache (disabled by default), after that, each thread uses its own DFA. This will increase memory usage to store thread local DFAs and redundant computation to build thread local DFAs (not too much). The benefit is that it can improve the concurrent performance running with multiple threads. In other words, when you find your concurent throughput is not high enough, you should consider turning on this option.
```

Comment:
Please follows Markdown best practices:

* Use line breaks after headers ([Heading Best Practices](https://www.markdownguide.org/basic-syntax/#heading-best-practices))
* Split long lines on several small ones (not longer than 120 chars). It makes git diffs much more convenient. ([Line Break Best Practices](https://www.markdownguide.org/basic-syntax/#line-break-best-practices)). At least split by sentences.

```suggestion
### Compile Aspects

When compiling generated files, you can configure a compile option according to your needs (also optional):

* A **thread local DFA macro**: Add `-DANTLR4_USE_THREAD_LOCAL_CACHE=1` to the compilation options
will enable using thread local DFA cache (disabled by default), after that, each thread uses its own DFA.
This will increase memory usage to store thread local DFAs and redundant computation to build thread local DFAs (not too much).
The benefit is that it can improve the concurrent performance running with multiple threads.
In other words, when you find your concurent throughput is not high enough, you should consider turning on this option.
```